homework6/task2_oop2/project2_oop2/oop2.py

```diff
@@ -0,0 +1,149 @@
+    """Raise when homework is inactive when its deadline has already ended."""
+
+
+class Person:
+    """Create an instance of a person. Base class for Student and Teacher classes."""
+
+    def __init__(self, first_name: str, last_name: str):
+        self.first_name = first_name
+        self.last_name = last_name
+
+
+class Student(Person):
+    """Create Student object."""
+
+    def do_homework(self, home_work: Homework, result: str) -> Any:
+        """Checks is the deadline of the given homework expired or not.
+
+        Return HomeworkResult class object if deadline hasn't expired yet, otherwise raise DeadlineError.
+        """
+        if home_work.is_active():
+            return HomeworkResult(self, home_work, result)
+        else:
+            raise DeadlineError("You are late")
+
+
+class HomeworkResult:
+    """Contain information about homework result.
+
+    Raise HomeworkError if a given homework is not a Homework class object.
+    """
+
+    def __init__(self, author: Student, homework: Homework, hw_result: str) -> None:
+        if not isinstance(homework, Homework):
+            raise HomeworkTypeError("You gave not a Homework object")
+        self.author = author
+        self.homework = homework
+        self.solution = hw_result
+        self.created = datetime.datetime.now()
+
+
+class HomeworkTypeError(Exception):
+    """Raise if a given homework in HomeworkResult constructor is not a Homework class object."""
+
+
+class Teacher(Person):
+    """Create Teacher object.
+
+    Has class attribute - homework_done, which is a data structure with dictionary interface. All HomeworkResult
+    objects are saved here after successful check_homework method applying.
+    """
+
+    homework_done = defaultdict(set)
+
+    @staticmethod
+    def create_homework(task: str, days_to_do_task: int) -> Homework:
+        """Creates an instance of Homework class object."""
+        return Homework(task, days_to_do_task)
+
+    @staticmethod
+    def check_homework(homework_result: HomeworkResult) -> bool:
+        """Check given homework result.
+
+        Return True if homework solution consists of more than 5 symbols, otherwise return False. If True, homework
+        solution is saved in homework_done data structure.
+        """
+        if len(homework_result.solution) > 5:
+            Teacher.homework_done[homework_result.homework.text].add(
+                homework_result.solution
+            )
+            return True
+        return False
+
+    @staticmethod
+    def reset_results(homework: Homework = None) -> None:
+        """Depending on a given argument, this method partly or fully remove results in homework_done data structure.
+
+        If homework is not given as an argument, this method remove all information in homework_done data structure.
+        Otherwise, it removes only results of a given homework.
+        """
+        if homework:
+            Teacher.homework_done[homework.text].clear()
```

Comment:
I'd prefer using `del Teacher.homework_done[homework.text]`  both approaches are work absolutely the same but have slightly different performance under specific cases.  `del` case is more general, and in my opinion better show your intention.